File: RxTools-library/src/main/java/com/vondear/rxtools/view/RxFrameLayoutTouch.java
package com.vondear.rxtools.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.FrameLayout;

/**
 * @author vondear
 * @date 2018/3/5
 * 解决滑动冲突
 */

public class RxFrameLayoutTouch extends FrameLayout {

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {

        if (ev.getAction() == MotionEvent.ACTION_DOWN) {
            getParent().requestDisallowInterceptTouchEvent(true);//拦截父类事件

        } else if (ev.getAction() == MotionEvent.ACTION_UP) {
            getParent().requestDisallowInterceptTouchEvent(false);
        }

        return super.dispatchTouchEvent(ev);
    }

    public RxFrameLayoutTouch(Context context) {
        super(context);
    }

    public RxFrameLayoutTouch(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public RxFrameLayoutTouch(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

}

File: RxUI/src/main/java/com/tamsiree/rxui/view/RxFrameLayoutTouch.kt
package com.tamsiree.rxui.view

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.FrameLayout

/**
 * @author tamsiree
 * @date 2018/3/5
 * 解决滑动冲突
 */
class RxFrameLayoutTouch : FrameLayout {
    override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
        if (ev.action == MotionEvent.ACTION_DOWN) {
            //拦截父类事件
            parent.requestDisallowInterceptTouchEvent(true)
        } else if (ev.action == MotionEvent.ACTION_UP) {
            parent.requestDisallowInterceptTouchEvent(false)
        }
        return super.dispatchTouchEvent(ev)
    }

    constructor(context: Context?) : super(context!!)
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr)
}